pythonforandroid/bootstraps/sdl2/build/src/main/java/org/libsdl/app/SDLActivity.java

```diff
@@ -2002,7 +2002,7 @@ public boolean onKeyPreIme (int keyCode, KeyEvent event) {
     public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
         ic = new SDLInputConnection(this, true);
 
-        outAttrs.inputType = InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD;
+        outAttrs.inputType = InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_NORMAL;
```

Comment:
Does ```TYPE_TEXT_VARIATION_NORMAL``` really have to be explicit? According to Android documentation, it is the default variation of ```TYPE_CLASS_TEXT```.

Reply:
Maybe, it doesn't have to be. I'll test it when I have time.

Reply:
Considering that:

- [TYPE_CLASS_TEXT](https://developer.android.com/reference/android/text/InputType.html#TYPE_CLASS_TEXT) equals 1.

-  [TYPE_TEXT_VARIATION_NORMAL](https://developer.android.com/reference/android/text/InputType.html#TYPE_TEXT_VARIATION_NORMAL) equals 0.

Then ```1 | 0 = 1```, so I think it's a good idea to remove ```TYPE_TEXT_VARIATION_NORMAL```:

```suggestion
        outAttrs.inputType = InputType.TYPE_CLASS_TEXT;
```

By the way, keyboards with predictive text have a non-desirable behavior with this PR: written text (containing just letters) does not appear in TextInput. Only symbols, numbers, and text selected from suggested words are visible.

Test it and tell me if you can reproduce the bug, please. I'll try to fix it.

Reply:
Considering that:

- [TYPE_CLASS_TEXT](https://developer.android.com/reference/android/text/InputType.html#TYPE_CLASS_TEXT) equals 1.

-  [TYPE_TEXT_VARIATION_NORMAL](https://developer.android.com/reference/android/text/InputType.html#TYPE_TEXT_VARIATION_NORMAL) equals 0.

Then ```1 | 0 = 1```, so I think it's a good idea to remove ```TYPE_TEXT_VARIATION_NORMAL```:

```suggestion
        outAttrs.inputType = InputType.TYPE_CLASS_TEXT;
```

By the way, keyboards with predictive text have a non-desirable behavior with this PR: written text (containing just letters) does not appear in TextInput. Only symbols, numbers, and text selected from suggested words are visible.

Test it and tell me if you can reproduce the bug, please. I'll try to fix it.

Reply:
I can reproduce the bug.

I do not see any difference in behavior with or without TYPE_TEXT_VARIATION_NORMAL

When prediction is provided, the text does not appear in the TextInput until it is selected from the bar or "confirmed" with space character. Only some keyboard, like Swiftkey, selects the word on space character, others do not (Gboard?).
In most of the cases, on backspace, the text is first deleted from this invisible "buffer" and then from the TextInput.
Symbols and numbers behavior varies depending on a keyboard and whether the keyboard tries to predict numeric input.

Swipe on Swiftkey keyboard inserts the word twice, Gboard does not inserts anything (or inserts a space character? maybe it has something to do with the space character behavior)